--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -191,6 +191,7 @@
 #define CONFIG_ENV_OVERWRITE
 
 /* Environment */
+#define CONFIG_ENV_IS_NOWHERE
 #ifndef CONFIG_ENV_IS_NOWHERE
 # ifndef CONFIG_SYS_NO_FLASH
 /* Environment in NOR flash */
@@ -219,6 +220,7 @@
 		"cp.b ${nor_flash_off} ${load_addr} ${fit_size} && " \
 		"bootm ${load_addr}\0" \
 	"sdboot=echo Copying FIT from SD to RAM... && " \
+		"setenv bootargs console=ttyPS0,115200 earlyprintk rdinit=/etc/preinit && " \
 		"load mmc 0 ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
